### DISCUSSION TITLE
SCRIPT_DEVANAGARI strange behavior

### DISCUSSION THREAD

**Discussion by rabbitUdomly** [harfbuzz/harfbuzz#3064] [Q&A] [answered]
Hello

I'm using harfbuzz to convert text to array glyphs codepoint.

direction = 'ltr'
script = SCRIPT_DEVANAGARI
inputStr = string to convert
hb_ft_font = my font

    hb_buffer_t *buf = hb_buffer_create();

    hb_buffer_set_direction(buf, direction);
    hb_buffer_set_script(buf, script);

    hb_buffer_add_utf8( buf, inputStr, nLength, 0, nLength );
    hb_shape(hb_ft_font, buf, NULL, 0);

    unsigned int glyph_count;
    hb_glyph_info_t *glyph_info = hb_buffer_get_glyph_infos(buf, &glyph_count);

	int it;
    for (it = 0; it < glyph_count; it++)
    {
        int glyph = glyph_info[it].codepoint;
        if (glyph) {
	        printf("%d\n", glyph);
        }
    }

I recently updated harfbuzz 1.7.2 to 2.6.4
Got a weird result:

1. When convert string with Umlaut and glyph charset "Ä aी" I am got codepoint for Ä like a 134
2. When convert string with Umlaut and without glyph charset "Ä aन" I am got codepoint for Ä like a 36, 386

I guess that if there is a glyph in the text, the shape parses the text differently, but in all cases I expect the same result.

If used Arabic script, result always Ä like a 134

Before that I used harfbuzz 1.7.2 and that worked fine. 

Perhaps there are some changes that affect this behavior, or am I doing something wrong?

**Comment by behdad** [comment, marked as answer]
You can't mix Latin and Devanagari and expect it to work.

**Reply by khaledhosny** [reply]
Like Behdad said, you need to itemize the text per-script and shape each script run separately. Or you can use a library like https://github.com/HOST-Oman/libraqm to do that for you.
